### PR TITLE
enable output to directory and search for files recursively

### PR DESCRIPTION
**Pull request by slinstaedt**
If you are interested... used this for creating configuration files during docker container creation.

### CODE REVIEW

File: templater.sh [hunk 1]
```diff
@@ -274,9 +283,13 @@ function main() {
     fi
     replace_ifs > /dev/null 2>&1
     render
-    echo "$TEMPLATE_CONTENT"
+    if [ -z "$output" ]; then
```
Comment (owenstranathan) [inline templater.sh]: I don't love that this disables echo by default, should be another option

File: templater.sh [hunk 2]
```diff
@@ -274,9 +283,13 @@ function main() {
...
+        echo "$TEMPLATE_CONTENT"
+    else
+        mkdir -p $(dirname "$output/$template_path")
```
Comment (owenstranathan) [inline templater.sh]: This subshell should be quoted to avoid word splitting

File: templater.sh [hunk 3]
```diff
@@ -285,13 +298,18 @@ if [[ -f "$template_path" ]]; then
...
     shopt -s globstar nullglob
-    templates=( "$template_path/"* )
-    len=${#templates[@]}
-    len=$((len-1))
-    for i in $(seq 0 $len); do
-        main "${templates[i]}"
-        if [ $i -lt $len ]; then
-        echo -e "$delimiter"
+    if [ "$recursive" == "true" ]; then
+        templates=$(find "$template_path")
```
Comment (owenstranathan) [inline templater.sh]: using this without `read -a` won't always work. prefer bash4 globstar

File: templater.sh [hunk 4]
```diff
@@ -285,13 +298,18 @@ if [[ -f "$template_path" ]]; then
...
+    else
+        templates="$template_path/"*
```
Comment (owenstranathan) [inline templater.sh]: this used to be `templates=( "$template_path/"* )` which made it an array.
Now it's just the first file from the glob. That broken :(

**Review by owenstranathan — CHANGES_REQUESTED** [review]
I made a PR with the changes I wanted to see you your fork. If you merge that into your fork then I'll merge it into mine 👍

**Comment by slinstaedt** [comment]
I am by far not an export for shell scripting, so any improvements are welcome.

**Comment by owenstranathan** [comment]
That's the beautiful thing about bash, it's so complex and terrible that we're all equal under it's umbrella of untenability.

**Pull request merged**